Update installation instructions to use a docker-compose override file

kellnerd opened the pull request:
In addition to what the title promises, this PR also includes a note about the approximate disk space requirements of the Docker images.
While I was already at it, I also run a code spell checker over the repository, fixed the markup of a broken link and made the formatting (of the installation document) more consistent in general, see the individual commits.

Please note that I did not test the example override file but one with additional overrides to disable Elasticsearch on my broken setup.
I previewed the rendered installation file with GitHub, which does not support all features of RTD, but at least helped to avoid the simplest formatting mistakes.

kellnerd committed:
Fix obvious typos found by a code spell checker

"searchForMatchingEditionGroups" also has to be fixed in the code itself.

kellnerd committed:
Update live reload instructions to use a docker-compose override file

kellnerd committed:
Fix a broken link and make the formatting more consistent

kellnerd committed:
Add a note about disk space requirements

kellnerd commented on docs/installation.rst:
```diff
@@ -145,14 +153,26 @@ For example:
       # 2. Mount the src directory
           - "./src:/home/bookbrainz/bookbrainz-site/src"
+
+Ideally you save this new Compose file inside the ``local/`` directory, which will be ignored by git, for example as ``local/docker-compose.live-reload.yml``.
```
This change still has to be done in the `.gitignore` of `bookbrainz-site` of course.
I haven't done this myself, as you said that you will do it yourself.

MonkeyDo replied:
Done !

kellnerd commented:
Oh, and I've just seen that this PR's content conflicts with #11 which added a new section that suggests to apply changes directly to `docker-compose.yml`.
There might be merge conflicts even because of my typo hunt and the formatting changes.

MonkeyDo commented:
> Oh, and I've just seen that this PR's content conflicts with #11 which added a new section that suggests to apply changes directly to `docker-compose.yml`. There might be merge conflicts even because of my typo hunt and the formatting changes.

No merge conflicts after merging it looks like, but I would appreciate if you gave this PR a once-over to make sure it hasn't been perturbed by merging #11

kellnerd committed:
Format installation headings

kellnerd committed:
Drop outdated section about git submodules (which are no longer used)

kellnerd committed:
Update debugging instructions to use a docker-compose override file

kellnerd commented:
So, I've now rebased the PR, fixed a few more formatting inconsistencies from #11 and adapted its new debugging section to also use a compose override (which I haven't tested myself!)

Additionally I also dropped the git submodules section, but left the git command itself untouched as that will be updated by #15.

kellnerd commented:
Hmm, somehow 4cc6709a14128f2c517a926d4325e66b8cdad940 is breaking the GitHub preview, not sure if that's a real issue or just GitHub? [The previous commit](https://github.com/kellnerd/bookbrainz-dev-docs/blob/21861238bb3ab11ec90e2a8f8b2addf4deb333d0/docs/installation.rst) still renders on GitHub...

MonkeyDo commented on docs/installation.rst:
```diff
@@ -19,28 +19,16 @@ Setting Up
-Currently we are using these submodules
-
-* `MonkeyDo/Lobes`_ in ``src/client/stylesheets/lobes``
+    git clone --recurse-submodules https://github.com/metabrainz/bookbrainz-site.git
```
We don't need this flag anymore as we stopped using git submodules
```suggestion
    git clone https://github.com/metabrainz/bookbrainz-site.git
```

See https://github.com/metabrainz/bookbrainz-site/commit/9373aa0ea64f7fbeb84dfcda6b6f1188fab7ad1d

MonkeyDo commented on docs/installation.rst:
```diff
@@ -264,23 +270,35 @@ For example:
+Ideally you save this new Compose file inside the ``local/`` directory, which will be ignored by git, for example as ``local/docker-compose.live-reload.yml``.
+
+Now you have to explicitly tell Docker-Compose which Compose files it should read when the ``./develop.sh`` script is run.
+In addition to the default ``docker-compose.live-reload.yml`` we also want Compose to read our custom file with the overrides.
```
Should this read "the default docker-compose.yml" instead of the live-reload override file?

kellnerd replied:
You're right of course, I vaguely remember having fixed that already, but I didn't find the changes in one of my two local clones. So I had probably only _intended_ to fix it after copy-pasting the markup from above (entering two consecutive backticks with a keyboard is a PITA, let alone having to do it twice because VS Code's feature to wrap selected text automatically did not work for this kind of quotes).
Anyway, I've fixed this line now and rebased onto the latest master.

MonkeyDo reviewed (commented):
Thanks a lot for the addition and the cleanup !

kellnerd committed:
Use reference markup instead of a descriptive reference

MonkeyDo reviewed (approved):
Thank you !
